core/src/commonMain/kotlin/Peripheral.kt

```diff
@@ -138,3 +140,19 @@ public interface Peripheral {
         onSubscription: OnSubscriptionAction = {},
     ): Flow<ByteArray>
 }
+
+/** Remember the bluetooth connected is not the same thing as Kable's [State.Connected]. */
+internal suspend fun Peripheral.suspendUntilConnectingServices() {
+    state.first { it == State.Connecting.Services }
+}
```

Comment:
Perhaps we should change this to?:

```suggestion
internal suspend fun Peripheral.suspendUntil(state: State) {
    state.first { it == state }
}
```

It barely does anything and almost feels we should just use `state.first` where we were calling this function, but I think the syntax sugar does make the code easier to read at the call-site.

But if we pass in the state it makes the code a bit more self documenting and we can get rid of a lot (if not all) of the KDoc suggestions I made, for example:

```kotlin
peripheral.suspendUntil(Connecting.Services)
```

Command clicking `Connecting.Services` can take you to the KDoc for it which would explain when that state is emitted (basically a re-write of the KDoc I suggested above.